### PR TITLE
Upmerge 17.11.2020: Synchronize repositories: zephyr, mcuboot, nfxlib, and mcumgr

### PR DESCRIPTION
**Pull request by ioannisg**
**Upmerge in-progress: Final testing phase.**

Manifest update for synchronizing zephyr and mcu-mgr.

SDK-ZEPHYR PR: http://github.com/nrfconnect/sdk-zephyr/pull/389
SDK-MCUMGR PR: https://github.com/nrfconnect/sdk-mcumgr/pull/12
SDK-MCUBOOT PR: https://github.com/nrfconnect/sdk-mcuboot/pull/130
SDK-NRFXLIB PR: https://github.com/nrfconnect/sdk-nrfxlib/pull/332

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: a3a6410b55fa97a9c6400c37d362c9137c5050b6
+      revision: pull/389/head
```
Comment (oyvindronningstad) [inline west.yml:53]: Update the name-whitelist to bring in trusted-firmware-m.

File: west.yml [hunk 2]
```diff
@@ -96,7 +96,7 @@ manifest:
       path: bootloader/mcuboot
     - name: mcumgr
       repo-path: sdk-mcumgr
-      revision: v0.1.0-ncs1
+      revision: f28a637db12c4b12fb2b18bddc6b2a0deaa95251
```
Comment (de-nordic) [inline west.yml:99]:
```suggestion
      revision: pull/12/head
```

Comment (de-nordic) [reply west.yml]: @ioannisg Sorry for committing the suggestion to your changes (the github interface is very ambigius for me). Pleas squash them to your commit.

File: west.yml (hunk 1 above)
Comment (ioannisg) [reply west.yml:53]: done now

**Comment by rlubos** [comment]
NCS OpenThread samples fail to link with the following:
```
/home/robert/zephyr-sdk/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: /home/robert/repos/zephyrproject/nrfxlib/mpsl/lib/cortex-m4/soft-float/libmpsl.a(libmpsl_debug_soft__obfuscated.elf): in function `nrf_802154_fal_tx_power_get':
:(.text+0x0): multiple definition of `nrf_802154_fal_tx_power_get'; modules/nordic/lib..__modules__hal__nordic.a(nrf_802154_sl_fem.c.obj):/home/robert/repos/zephyrproject/modules/hal/nordic/drivers/nrf_802154_sl_opensource/src/nrf_802154_sl_fem.c:137: first defined here
collect2: error: ld returned 1 exit status
```
I don't think we should use `nrf_802154_sl_opensource` in NCS, @hubertmis do we miss something in `sdk-nrf/nrfxlib`?

**Comment by hubertmis** [comment]
> I don't think we should use `nrf_802154_sl_opensource` in NCS, @hubertmis do we miss something in `sdk-nrf/nrfxlib`?

That's correct. @jciupis prepares a PR to nrfxlib

**Comment by hubertmis** [comment]
@rlubos , as a workaround you can remove MPSL from applications including 802.15.4. Multiprotocol won't work, but basic support should be present until nrfxlib is updated.

**Comment by ioannisg** [comment]
> > I don't think we should use `nrf_802154_sl_opensource` in NCS, @hubertmis do we miss something in `sdk-nrf/nrfxlib`?
> 
> That's correct. @jciupis prepares a PR to nrfxlib

@hubertmis @jciupis could you link here that PR?

**Comment by jciupis** [comment]
@ioannisg Here's the nrfxlib PR: https://github.com/nrfconnect/sdk-nrfxlib/pull/332 and its corresponding manifest update: https://github.com/nrfconnect/sdk-nrf/pull/3365

**Review by trond-snekvik — APPROVED** [review]

**Comment by jciupis** [comment]
@ioannisg I pushed the sdk-nrf/nrfxlib manifest update here to resolve missing dependencies both in this PR (as pointed out by @rlubos) and in the related nrfxlib PR

**Comment by jhn-nordic** [comment]
Mcuboot overflows 48k limit on Thingy:91 when compiling asset tracker for thingy91_nrf9160ns There is also issues with a I2C2 config in the downstream board definition.

**Comment by rlubos** [comment]
`asset_tracker`, `lwm2m_client` on nrf9160dk_nrf9160ns don't boot:
```
�*** Booting Zephyr OS build v2.4.0-ncs1-1693-ge273cda46b75  ***t
[00:00:00.333,587] <inf> mcuboot: Starting bootloader
[00:00:00.339,782] <inf> mcuboot: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
[00:00:00.349,792] <inf> mcuboot: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
[00:00:00.359,924] <inf> mcuboot: Boot source: none
[00:00:00.365,447] <inf> mcuboot: Swap type: none
[00:00:00.731,994] <inf> mcuboot: Bootloader chainload address offset: 0x10000
�00:00:00.739,685] <inf> mcuboot: Jumping to the first image slot
... and nothing happens ...
```

Zigbee samples does not build (FYI @wbober @tomchy):
```
/home/robert/repos/zephyrproject/nrf/subsys/partition_manager/pm.yml.zboss:2:10: fatal error: devicetree_legacy_unfixed.h: No such file or directory
    2 | #include <devicetree_legacy_unfixed.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Review by grochu — APPROVED** [review]

**Comment by jtguggedal** [comment]
> `asset_tracker`, `lwm2m_client` on nrf9160dk_nrf9160ns don't boot:
> 
> ```
> �*** Booting Zephyr OS build v2.4.0-ncs1-1693-ge273cda46b75  ***t
> [00:00:00.333,587] <inf> mcuboot: Starting bootloader
> [00:00:00.339,782] <inf> mcuboot: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
> [00:00:00.349,792] <inf> mcuboot: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
> [00:00:00.359,924] <inf> mcuboot: Boot source: none
> [00:00:00.365,447] <inf> mcuboot: Swap type: none
> [00:00:00.731,994] <inf> mcuboot: Bootloader chainload address offset: 0x10000
> �00:00:00.739,685] <inf> mcuboot: Jumping to the first image slot
> ... and nothing happens ...
> ```
> 
> Zigbee samples does not build (FYI @wbober @tomchy):
> 
> ```
> /home/robert/repos/zephyrproject/nrf/subsys/partition_manager/pm.yml.zboss:2:10: fatal error: devicetree_legacy_unfixed.h: No such file or directory
>     2 | #include <devicetree_legacy_unfixed.h>
>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> compilation terminated.
> ```

At least for the asset tracker, there's a bus fault in the SPM startup code on this instruction:
`bl z_arm_prep_c`

**Comment by mbolivar-nordic** [comment]
@hakonfam I looked through the DT macros used by the various pm.yml files, and it looks like they're just aligning everything to the block size of a partition:

```
$ rg -l devicetree_legacy_unfixed | xargs grep DT_
nrf/subsys/partition_manager/pm.yml.zboss:  placement: {after: [app], align: {start: DT_FLASH_ERASE_BLOCK_SIZE}}
nrf/samples/bootloader/pm.yml:    align: {start: DT_FLASH_ERASE_BLOCK_SIZE}
nrf/samples/bootloader/pm.yml:    align: {start: DT_FLASH_ERASE_BLOCK_SIZE}
nrf/scripts/partition_manager/partition_manager.rst:       align: {start: DT_FLASH_ERASE_BLOCK_SIZE}
nrf/scripts/partition_manager/partition_manager.rst:       align: {start: DT_FLASH_ERASE_BLOCK_SIZE}
```

Is it really necessary to specify this? Why doesn't partition manager always align partitions along partition erase block boundaries?

**Comment by hakonfam** [comment]
> > `asset_tracker`, `lwm2m_client` on nrf9160dk_nrf9160ns don't boot:
> > ```
> > �*** Booting Zephyr OS build v2.4.0-ncs1-1693-ge273cda46b75  ***t
> > [00:00:00.333,587] <inf> mcuboot: Starting bootloader
> > [00:00:00.339,782] <inf> mcuboot: Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
> > [00:00:00.349,792] <inf> mcuboot: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
> > [00:00:00.359,924] <inf> mcuboot: Boot source: none
> > [00:00:00.365,447] <inf> mcuboot: Swap type: none
> > [00:00:00.731,994] <inf> mcuboot: Bootloader chainload address offset: 0x10000
> > �00:00:00.739,685] <inf> mcuboot: Jumping to the first image slot
> > ... and nothing happens ...
> > ```

You need this PR https://github.com/nrfconnect/sdk-nrf/pull/3380
Or something like it which sets the INIT_HW config in SPM.

> > 
> > Zigbee samples does not build (FYI @wbober @tomchy):
> > ```
> > /home/robert/repos/zephyrproject/nrf/subsys/partition_manager/pm.yml.zboss:2:10: fatal error: devicetree_legacy_unfixed.h: No such file or directory
> >     2 | #include <devicetree_legacy_unfixed.h>
> >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> > compilation terminated.
> > ```

I added a fix for this in this pr https://github.com/nrfconnect/sdk-nrf/pull/3368

> At least for the asset tracker, there's a bus fault in the SPM startup code on this instruction:
> `bl z_arm_prep_c`

You need this PR https://github.com/nrfconnect/sdk-nrf/pull/3380 will fix it. The problem is that you are not resetting SPLIM, neither is MCUBoot.

**Review by hakonfam — CHANGES_REQUESTED** [review]
BLE samples are broken on nRF53. This is being looked at

**Review by grochu — COMMENTED** [review]
Need to do testing on Yoda DKs, seems like there is some issue with the Bluetooth samples.

**Review by grochu — CHANGES_REQUESTED** [review]
Removing the approval till BLE runs well on nRF53 DKs.

**Review by nvlsianpu — CHANGES_REQUESTED** [review]
looks like something is wrong around PM - testing smp_svr leds to application lost instead of update. Will investigate further

**Review by lmaciejonczyk — APPROVED** [review]

**Comment by de-nordic** [comment]
> 
> 
> looks like something is wrong around PM - testing smp_svr leds to application lost instead of update. Will investigate further

I will also look into it as I am quite sure that it worked yesterday.

**Review by lmaciejonczyk — CHANGES_REQUESTED** [review]
~~Multiprotocol stoped working for OpenThread samples~~
Actually we have problem with one variant (MTD) of our sample: coap_client

**Comment by de-nordic** [comment]
> 
> 
> > looks like something is wrong around PM - testing smp_svr leds to application lost instead of update. Will investigate further
> 
> I will also look into it as I am quite sure that it worked yesterday.

I can confirm @nvlsianpu  finding: smp_svr is overwriting itself.

**Comment by hakonfam** [comment]
@nvlsianpu can you try this https://github.com/ioannisg/sdk-nrf/pull/2 ?

**Review by ru-fu — APPROVED** [review]

**Review by b-gent — APPROVED** [review]

**Comment by MarekPorwisz** [comment]
> ~Multiprotocol stoped working for OpenThread samples~
> Actually we have problem with one variant (MTD) of our sample: coap_client

We did some digging with @lmaciejonczyk on this issue and it seems that this is a conflict with logs and uart power management. Possibly other applications may be affected by this as well.
This specific sample switches uart off when entering sleepy end device mode. Minimal log maps directly to printk. Printk does not verify if the device is in off state nor does the driver. Uart driver puts the character to tx queue and enters a loop until the sending is complete. This ends up in the infinite loop waiting for send to be finished on a disabled uart.

**Review by hakonfam — APPROVED** [review]
Verified booting, updates etc. @grochu  will handle the failing BLE so I don't NACK for that.

**Review by grochu — APPROVED** [review]
As discussed offline, the Bluetooth samples on nRF5340 can be fixed later, so I am unblocking the PR. @carlescufi, @ioannisg does this sound OK?

**Comment by ioannisg** [comment]
> Mcuboot overflows 48k limit on Thingy:91 when compiling asset tracker for thingy91_nrf9160ns There is also issues with a I2C2 config in the downstream board definition.

@nvlsianpu @de-nordic i wonder if this is due to the unification of the mcuboot partition size... or just things have changed. Can you comment?

**Comment by jtguggedal** [comment]
@hakonfam The asset tracker still doesn't boot, even with the fix from #3380 . Any ideas?

**Comment by rlubos** [comment]
@jtguggedal From what I see `asset_tacker` has it's own spm.conf file, I assume we need a fix from https://github.com/nrfconnect/sdk-nrf/pull/3357/commits/5761138da1a8e03da8235ad7001f709456a11f70 in there as well.

**Comment by ioannisg** [comment]
> @jtguggedal From what I see `asset_tacker` has it's own spm.conf file, I assume we need a fix from [5761138](https://github.com/nrfconnect/sdk-nrf/commit/5761138da1a8e03da8235ad7001f709456a11f70) in there as well.

This makes absolute sense, thanks @rlubos . If the SPM is loaded by MCUboot, we need to clear the ARCH code, to avoid faults during boot.

**Comment by ioannisg** [comment]
> > @jtguggedal From what I see `asset_tacker` has it's own spm.conf file, I assume we need a fix from [5761138](https://github.com/nrfconnect/sdk-nrf/commit/5761138da1a8e03da8235ad7001f709456a11f70) in there as well.
> 
> This makes absolute sense, thanks @rlubos . If the SPM is loaded by MCUboot, we need to clear the ARCH code, to avoid faults during boot.

@rlubos I merged the SPM.conf fix, please take another look

**Comment by jtguggedal** [comment]
@rlubos @ioannisg Good catch, thanks for fixing! The asset tracker now boots on nRF9160 DK. I'm not able to test cloud connectivity because of downtime on AWS IoT, but at least the app boots.

For Thingy91, I've posted a PR for Kconfig error: https://github.com/ioannisg/sdk-nrf/pull/4
The MCUboot flash partition overflow still persists, and has increased from 160 bytes to  448.
I switched off logging in MCUboot, and then the application builds and boots. Same cloud connectivity issue as DK, but that's expected while AWS IoT is down.

**Comment by maciekfabia** [comment]
> Zigbee samples does not build (FYI @wbober @tomchy):
> ```
> /home/robert/repos/zephyrproject/nrf/subsys/partition_manager/pm.yml.zboss:2:10: fatal error: devicetree_legacy_unfixed.h: No such file or directory
>     2 | #include <devicetree_legacy_unfixed.h>
>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> compilation terminated.
> ```

@rlubos, @ioannisg zigbee samples build and work correctly now

**Review by maciekfabia — APPROVED** [review]

**Comment by hakonfam** [comment]
> The MCUboot flash partition overflow still persists, and has increased from 160 bytes to 448.
> I switched off logging in MCUboot, and then the application builds and boots. Same cloud connectivity issue as DK, but that's expected while AWS IoT is down.

You can try using `CONFIG_LOG_MINIMAL` to still get logging, but a ~4k smaller footprint

**Comment by de-nordic** [comment]
> 
> For Thingy91, I've posted a PR for Kconfig error: [ioannisg#4](https://github.com/ioannisg/sdk-nrf/pull/4)
> The MCUboot flash partition overflow still persists, and has increased from 160 bytes to 448.
> I switched off logging in MCUboot, and then the application builds and boots. Same cloud connectivity issue as DK, but that's expected while AWS IoT is down.

You may also want to enable `CONFIG_SIZE_OPTIMIZATIONS`, if you haven't done so yet.

**Comment by ioannisg** [comment]
> @rlubos @ioannisg Good catch, thanks for fixing! The asset tracker now boots on nRF9160 DK. I'm not able to test cloud connectivity because of downtime on AWS IoT, but at least the app boots.
> 
> For Thingy91, I've posted a PR for Kconfig error: [ioannisg#4](https://github.com/ioannisg/sdk-nrf/pull/4)
> The MCUboot flash partition overflow still persists, and has increased from 160 bytes to 448.
> I switched off logging in MCUboot, and then the application builds and boots. Same cloud connectivity issue as DK, but that's expected while AWS IoT is down.

Thanks, this is now merged to the upmerge PR.

**Comment by ioannisg** [comment]
> As discussed offline, the Bluetooth samples on nRF5340 can be fixed later, so I am unblocking the PR. @carlescufi, @ioannisg does this sound OK?

It is OK for me.

**Review by nvlsianpu — APPROVED** [review]

**Comment by jtguggedal** [comment]
> > For Thingy91, I've posted a PR for Kconfig error: [ioannisg#4](https://github.com/ioannisg/sdk-nrf/pull/4)
> > The MCUboot flash partition overflow still persists, and has increased from 160 bytes to 448.
> > I switched off logging in MCUboot, and then the application builds and boots. Same cloud connectivity issue as DK, but that's expected while AWS IoT is down.
> 
> You may also want to enable `CONFIG_SIZE_OPTIMIZATIONS`, if you haven't done so yet.

Thanks for the tip! It worked and I have created a PR to the MCUboot upmerge branch.

**Comment by jtguggedal** [comment]
There's a build warning for the datetime library:
```
[209/289] Building C object modules/nrf/lib/date_time/CMakeFiles/..__nrf__lib__date_time.dir/date_time.c.obj
ncs/nrf/lib/date_time/date_time.c:467:13: warning: Macro is deprecated
  467 |      CONFIG_APPLICATION_INIT_PRIORITY);
      |             ^~~~~~~~~~~~~~~~~~~~~
```

Any ideas what should be used instead of that deprecated macro to fix the warning @rlubos ?

**Comment by rlubos** [comment]
@jtguggedal It seems that `DEVICE_INIT` was deprecated:
https://github.com/zephyrproject-rtos/zephyr/blame/77fb9b8b2c173c8bb04fc96c934d170743753440/doc/releases/release-notes-2.5.rst#L55
The release note suggests to use `DEVICE_DEFINE`, but to be honest I'm not sure why is the `DEVICE_*` macro used there in the first place. I'd consider `SYS_INIT` instead.

**Comment by maciekfabia** [comment]
There's OTA issue in zigbee samples, we're investigating it.

**Comment by simensrostad** [comment]
> @jtguggedal It seems that `DEVICE_INIT` was deprecated:
> https://github.com/zephyrproject-rtos/zephyr/blame/77fb9b8b2c173c8bb04fc96c934d170743753440/doc/releases/release-notes-2.5.rst#L55
> The release note suggests to use `DEVICE_DEFINE`, but to be honest I'm not sure why is the `DEVICE_*` macro used there in the first place. I'd consider `SYS_INIT` instead.

Build warning fixed in https://github.com/ioannisg/sdk-nrf/pull/5.

**Comment by grochu** [comment]
@ioannisg wouldn't it be possible to get this commit in as part of the upmerge: https://github.com/zephyrproject-rtos/zephyr/commit/19a93a3c68c79dd920678f38fbfcfd66fe32afee? We need this DTS description to support Eagle in Radio test examples.

**Comment by ioannisg** [comment]
> @ioannisg wouldn't it be possible to get this commit in as part of the upmerge: [zephyrproject-rtos/zephyr@19a93a3](https://github.com/zephyrproject-rtos/zephyr/commit/19a93a3c68c79dd920678f38fbfcfd66fe32afee)? We need this DTS description to support Eagle in Radio test examples.

@grochu can you send a PR against the upmerge PR?

**Comment by grochu** [comment]
> @grochu can you send a PR against the upmerge PR?

I'll let @KAGA164 do this.

**Comment by ioannisg** [comment]
> > @jtguggedal It seems that `DEVICE_INIT` was deprecated:
> > https://github.com/zephyrproject-rtos/zephyr/blame/77fb9b8b2c173c8bb04fc96c934d170743753440/doc/releases/release-notes-2.5.rst#L55
> > The release note suggests to use `DEVICE_DEFINE`, but to be honest I'm not sure why is the `DEVICE_*` macro used there in the first place. I'd consider `SYS_INIT` instead.
> 
> Build warning fixed in [ioannisg#5](https://github.com/ioannisg/sdk-nrf/pull/5).

Merged, thanks @simensrostad

**Review by de-nordic — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

**Review by oyvindronningstad — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]

**Comment by ioannisg** [comment]
> > @grochu can you send a PR against the upmerge PR?
> 
> I'll let @KAGA164 do this.
> This is a PR to nrfconnect/sdk-zephyr: [ioannisg/fw-nrfconnect-zephyr-1#2](https://github.com/ioannisg/fw-nrfconnect-zephyr-1/pull/2)

As I mentioned in the PR, this looks like the patch merged upstream, so I'd prefer to merge this after the upmerge is completed.

**Review by lmaciejonczyk — APPROVED** [review]

**Comment by pdunaj** [comment]
Regarding the nrf_desktop the following targets fail to build:
nrf52810dmouse_nrf52810 Debug
nrf52820dongle_nrf52820 ReleaseMCUBoot
nrf52833dongle_nrf52833 DebugMCUBoot
nrf52833dongle_nrf52833 ReleaseMCUBoot

The first one is a RAM overflow but the subsequent have some issue in MCUBoot config.
```
warning: attempt to assign the value 'y' to the undefined symbol SINGLE_IMAGE_DFU
```

**Comment by pdunaj** [comment]
> The first one is a RAM overflow but the subsequent have some issue in MCUBoot config.
> 
> ```
> warning: attempt to assign the value 'y' to the undefined symbol SINGLE_IMAGE_DFU
> ```

Seems that the option was renamed to `SINGLE_APPLICATION_SLOT`.

**Comment by pdunaj** [comment]
> > The first one is a RAM overflow but the subsequent have some issue in MCUBoot config.
> > ```
> > warning: attempt to assign the value 'y' to the undefined symbol SINGLE_IMAGE_DFU
> > ```
> 
> Seems that the option was renamed to `SINGLE_APPLICATION_SLOT`.

After fixing this there are more issues with build warnings:
```
mcuboot/boot/zephyr/flash_map_extended.c:59:20: warning: implicit declaration of function 'FLASH_AREA_IMAGE_PRIMARY'; did you mean 'FLASH_AREA_IMAGE_SCRATCH'?
```

And the image overflows and fails to link.
We can focus on fixing the overflow later on - probably even before the next NCS release - but this warning does not seem right.

**Comment by pdunaj** [comment]
I have checked nrf_desktop using mouse configuration for nrf52840dk. Basic functionality seems working correct.

Hi @wiminordic , do you know if this PR has executed our CI (the extended one)?

**Comment by pdunaj** [comment]
> this warning does not seem right

Seems the rename is not complete and in some places there is still SINGLE_IMAGE_DFU instead of SINGLE_APPLICATION_SLOT. Changing names fixes building but it seem the mcuboot PR is broken.

**Review by pdunaj — APPROVED** [review]
Seems ok. The overflow will be handled later before the release.
I suggest checking with @wiminordic if all tests passed. But if nothing is wrong I expect CI hooks to execute everything that is needed.

**Review by hurricaneFromMoscow — APPROVED** [review]

**Comment by maciekfabia** [comment]
> There's OTA issue in zigbee samples, we're investigating it.

@ioannisg The OTA issue doesn't occure any more. Everything is ok regarding zigbee.

**Review by joerchan — APPROVED** [review]

**Review by jhn-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
Good work 👍

**Comment by tejlmand** [comment]
All SHAs verified in manifest.
Looks good.

**Review by utzig — APPROVED** [review]

**Comment by hakonfam** [comment]
Great job @ioannisg  🙌

**Pull request merged**